cardano-api/src/Cardano/Api/Utils.hs

```diff
@@ -50,3 +70,29 @@ failEither = either fail pure
 
 failEitherWith :: MonadFail m => (e -> String) -> Either e a -> m a
 failEitherWith f = either (fail . f) pure
+
+note :: MonadFail m => String -> Maybe a -> m a
+note msg = \case
+  Nothing -> fail msg
+  Just a -> pure a
+
+parseFilePath :: String -> String -> Opt.Parser FilePath
```

Comment:
All of these do not belong in Cardano.Api. They should live in cadano-cli.

Reply:
yeah, these could be moved until a cardano-cli utils file